tests/syntax/test_errors.py

```diff
@@ -22,126 +23,103 @@ def test_bad_extension(tmpdir):
     with pytest.raises(RuntimeError), open(path, 'w'):
         scenic.scenarioFromFile(path)
 
-### Parse errors during token translation
-
-## Operators used in Python but not Scenic
-
-@pytest.mark.parametrize('op', ('~',))
-def test_illegal_unary_operators(op):
-    with pytest.raises(TokenParseError):
-        compileScenic(f'x = {op}2')
-
-@pytest.mark.parametrize('op', ('<<', '>>', '|', '&', '^', '//'))
-def test_illegal_binary_operators(op):
-    # Basic usage
-    with pytest.raises(TokenParseError):
-        compileScenic(f'x = 3 {op} 2')
-    # Augmented assignments
-    with pytest.raises(TokenParseError):
-        compileScenic(f'x {op}= 4')
-
-def test_illegal_statements():
-    with pytest.raises(ScenicSyntaxError):
-        compileScenic("""
-            async def foo(): pass
-            ego = Object
-        """)
+### Parse errors
 
 ## Constructor definitions
 
 badNames = ('', '3', '+', 'Behavior')
 
 @pytest.mark.parametrize('name', badNames)
 def test_illegal_constructor_name(name):
-    with pytest.raises(TokenParseError):
-        compileScenic(f'constructor {name}:\n' '    pass')
+    with pytest.raises(ScenicSyntaxError):
+        compileScenic(f'class {name}:\n' '    pass')
 
 @pytest.mark.parametrize('name', badNames)
 def test_illegal_constructor_superclass(name):
-    with pytest.raises(TokenParseError):
-        compileScenic(f'constructor Foo({name}):\n' '    pass')
+    with pytest.raises(ScenicSyntaxError):
+        compileScenic(f'class Foo({name}):\n' '    pass')
 
-def test_constructor_python_superclass():
-    with pytest.raises(TokenParseError):
-        compileScenic('constructor Foo(object):\n' '    pass')
+def test_malformed_constructor():
+    with pytest.raises(ScenicSyntaxError):
+        compileScenic('class Foo\n' '    pass')
+    with pytest.raises(ScenicSyntaxError):
+        compileScenic('class Foo(Bar:\n' '    pass')
 
-def test_constructor_undefined_superclass():
-    with pytest.raises(TokenParseError):
-        compileScenic('constructor Foo(Bar):\n' '    pass')
+def test_multiple_inheritance():
+    with pytest.raises(ScenicSyntaxError):
+        compileScenic("class Bad(Object, Point):\n    pass")
```

Comment:
This code should not raise an error any more. It turns out the test is passing because `compileScenic` is removing the indentation of the body of the class, yielding a syntax error (we didn't get this problem earlier because `compileScenic` did not originally use `indent.cleandoc` to support multi-line strings). All the tests in this file which use `compileScenic` on multi-line scenarios should be adjusted to use a multi-line string.

Reply:
Good catch, this revealed some broken behavior.